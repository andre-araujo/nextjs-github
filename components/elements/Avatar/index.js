import { string } from 'prop-types';

import {
    Wrapper,
    Image,
} from './styles';

const Avatar = ({
    url,
}) => (
    <Wrapper>
        <Image
            alt="user avatar"
            src={url}
        />
    </Wrapper>
);

Avatar.propTypes = {
    url: string.isRequired,
};

export default Avatar;
