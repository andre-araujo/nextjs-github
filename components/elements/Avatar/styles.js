import styled from 'styled-components';

export const Wrapper = styled.div`
    text-align: center;
`;

export const Image = styled.img`
    width: 100%;
    height: auto;
    border-radius: 5%;
`;
