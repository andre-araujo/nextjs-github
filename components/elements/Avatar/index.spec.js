import React from 'react';
import { shallow } from 'enzyme';

import Avatar from './index';

describe('Avatar component', () => {
    it('should match snapshot with url as something', () => {
        expect(shallow(
            <Avatar url="something" />,
        )).toMatchSnapshot();
    });

    it('should match snapshot with url as testing', () => {
        expect(shallow(
            <Avatar url="testing" />,
        )).toMatchSnapshot();
    });
});
