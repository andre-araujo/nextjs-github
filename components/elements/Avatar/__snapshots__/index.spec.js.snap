// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component should match snapshot with url as something 1`] = `
<styled.div>
  <styled.img
    alt="user avatar"
    src="something"
  />
</styled.div>
`;

exports[`Avatar component should match snapshot with url as testing 1`] = `
<styled.div>
  <styled.img
    alt="user avatar"
    src="testing"
  />
</styled.div>
`;
